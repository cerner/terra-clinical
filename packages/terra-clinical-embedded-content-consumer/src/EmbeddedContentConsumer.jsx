import React from 'react';
import PropTypes from 'prop-types';
import { Consumer } from 'xfc';
import './EmbeddedContentConsumer.scss';

const propTypes = {
  /**
   * The source URL of the content to load.
   */
  src: PropTypes.string.isRequired,
  /**
   * An object containing event handlers keyed by the event name.
   * (e.g. {'eventA': function() {}, 'eventB': function() {}})
   */
  eventHandlers: PropTypes.object,
};

const defaultProps = {
  src: undefined,
  eventHandlers: {},
};

class EmbeddedContentConsumer extends React.Component {

  componentDidMount() {
    if (this.embeddedContentWrapper) {
      Consumer.init();
      this.xfcFrame = Consumer.mount(this.embeddedContentWrapper, this.props.src);

      if (this.props.eventHandlers) {
        Object.keys(this.props.eventHandlers).forEach((event) => {
          this.xfcFrame.on(event, this.props.eventHandlers[event]);
        });
      }
    }
  }

  render() {
    return (
      <div
        className="terraClinical-EmbeddedContentConsumer"
        ref={(element) => { this.embeddedContentWrapper = element; }}
      />
    );
  }
}

EmbeddedContentConsumer.propTypes = propTypes;
EmbeddedContentConsumer.defaultProps = defaultProps;

export default EmbeddedContentConsumer;
