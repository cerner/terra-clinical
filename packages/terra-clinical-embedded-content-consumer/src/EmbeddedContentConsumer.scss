@import '~terra-mixins';

.terraClinical-EmbeddedContentConsumer {
  height: 100%;
  width: 100%;

  .xfc { // stylelint-disable-line selector-class-pattern
    height: 100%;
    width: 100%;

    iframe {
      border: 0;
      height: 100%;
      width: 100%;
    }
  }
}
