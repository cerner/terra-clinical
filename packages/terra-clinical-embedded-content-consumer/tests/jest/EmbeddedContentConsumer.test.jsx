import React from 'react';
import EmbeddedContentConsumer from '../../lib/EmbeddedContentConsumer';

// Snapshot Tests
it('should render the terraClinical-EmbeddedContentConsumer wrapper', () => {
  const embeddedContentConsumer = <EmbeddedContentConsumer src="https://www.google.com/" />;

  const wrapper = shallow(embeddedContentConsumer);
  expect(wrapper).toMatchSnapshot();
});

it('should validate the inputs', () => {
  const src = 'https://www.google.com/';
  const handlers = [{ key: 'eventKey', handler: () => {} }];

  const embeddedContentConsumer = (<EmbeddedContentConsumer
    src={src}
    handlers={handlers}
  />);
  const wrapper = mount(embeddedContentConsumer);

  expect(wrapper.props().src).toBe(src);
  expect(wrapper.props().handlers).toBe(handlers);
});
