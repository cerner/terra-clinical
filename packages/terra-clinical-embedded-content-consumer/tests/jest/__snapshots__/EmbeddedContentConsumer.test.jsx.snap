exports[`test should render the terraClinical-EmbeddedContentConsumer wrapper 1`] = `
<div
  className="terraClinical-EmbeddedContentConsumer" />
`;
